### DISCUSSION TITLE
Github API for deleting issues

### DISCUSSION THREAD

**Discussion by thilllon** [community/community#39520] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Can't find api for deleting issues.
Is there any plan to add the API for it?
https://octokit.github.io/rest.js/v19#issues-create
https://docs.github.com/en/rest/issues/issues

**Comment by LangLangBart** [comment, marked as answer]
You are right, no [GitHub Docs - REST API](https://docs.github.com/en/rest/issues) to delete an issue.

---

Alternative way: [GitHub Docs - GraphQL API - mutation:deleteissue](https://docs.github.com/en/graphql/reference/mutations#deleteissue)

#### query
```graphql
mutation {
  deleteIssue(input: {issueId: "I_kwDOIV8V6M5WM_mv", clientMutationId: "test issue delete"}) {
    clientMutationId
  }
}
```

#### result
```json
{
  "data": {
    "deleteIssue": {
      "clientMutationId": "test issue delete"
    }
  }
}
```

**Comment by myklst** [comment]
Hello, any updates of this issue? I would like to delete issue by API not GraphQL.

**Comment by Evander-Greer** [comment]
GitHub API and Issue Deletion:

Deleting Issues with GitHub API:
Unfortunately, GitHub's REST API does not allow the deletion of issues directly. As of now, GitHub does not support an API endpoint to delete issues once they have been created.

Why No Issue Deletion?

GitHub generally does not allow deletion of issues as part of the best practices for version control and maintaining history. Deleting issues could lead to loss of context for conversations, discussions, and tracking for projects, which is why it's restricted.

However, issues can be closed, which is typically sufficient to mark them as no longer relevant or active.

What You Can Do:

While deleting issues is not possible, there are other workarounds you can consider:

Close the issue: You can close issues that are no longer relevant using the GitHub API. This is the recommended way to handle issues that no longer need attention.

Label issues as "Won't Fix" or use a custom label such as "Deleted" to mark issues that you don’t want to track anymore.

Comment on issues: You could add a comment explaining that the issue was deleted or is no longer relevant if that’s important for transparency.

GitHub API for Managing Issues:
GitHub provides a wide array of operations for managing issues through their REST API:

Create: Create an issue

Edit: Update an issue

Close: Close an issue

List: List issues for a repository

Additional Resources:

GitHub API Documentation: If you need further details, you can explore more from the [GitHub REST API documentation](https://docs.github.com/en/rest/issues/issues)
.

For working with GitHub issues programmatically, Octokit (the official GitHub API client for JavaScript) is a great tool, but it still lacks a method for deleting issues.
Best regards.
